client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -200,6 +211,20 @@ private void handleEndOfSegment(SegmentInputStream oldSegment) throws Reinitiali
             throw e;
         }
     }
+    
+    private void handleSegmentTruncated(SegmentInputStream segmentReader) throws ReinitializationRequiredException {
+        Segment segmentId = segmentReader.getSegmentId();
+        log.info("{} encountered truncation for segment {} ", this, segmentId);
+        @Cleanup
+        SegmentMetadataClient metadataClient = metadataClientFactory.createSegmentMetadataClient(segmentId);
+        try {
+            long startingOffset = metadataClient.getSegmentInfo().getStartingOffset();
+            segmentReader.setOffset(startingOffset);
```

Comment:
Do we really need to add the metadata factory the constructor only to set the offset here? Could we have a new call in `SegmentInputStream` instead that enables us to set the offset?

Reply:
Per our discussion, delaying this until: #2197